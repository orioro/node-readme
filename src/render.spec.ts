import * as fs from 'fs'
import * as path from 'path'
import {
  render
} from './render'

const FIXTURES_PATH = path.join(__dirname, '../test')
const TMP_PATH = path.join(__dirname, '../tmp')

describe('render', () => {
  test('', () => {
    return render({
      templates: ['**/.*.md'],
      comments: ['**/*.ts'],
      cwd: FIXTURES_PATH
    })
    .then(() => {
      expect(fs.readFileSync(path.join(FIXTURES_PATH, 'fixture-1/readme.md'), 'utf8'))
        .toMatchSnapshot()
    })
  })
})
