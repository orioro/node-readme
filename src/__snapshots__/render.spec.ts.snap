// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render  1`] = `
"# readme

Helper utility to generate \`readme.md\` files for projects.
Parses source code looking for JSDoc-like comments.

## API

##### \`func1(strParam = '', numParam = 0)\`

Lorem ipsum dolor sit amet, consectetur adipiscing elit,
sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
pariatur.

\`\`\`js
const result = func1('some string', 10)
// false
\`\`\`

Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

- \`strParam\` {string} Sed ut perspiciatis, unde omnis iste natus error
                         sit voluptatem accusantium doloremque laudantium. Default: \`''\`
- \`numParam\` {number} Nemo enim ipsam voluptatem, quia voluptas
                         sit, aspernatur aut odit aut fugit. Default: \`0\`
- Returns: \`result\` {boolean} Nam libero tempore, cum soluta nobis est
                         eligendi optio.

##### \`func2(param1, param2)\`

Aliquam at metus molestie, luctus arcu id, ornare ligula.

- \`param1\` {Object}
- \`param1.opt1\` {string} Some option
- \`param1.opt2\` {string} Some other option
- \`param2\` {number}
- Returns: \`result\` {CustomType}

##### \`someMethod(param)\`

- \`param\` {number} Excepteur sint occaecat cupidatat non proident
- Returns: \`result\` {number} Lorem ipsum dolor sit amet, consectetur adipiscing
"
`;
